src/utils/withPromise.ts

```diff
@@ -0,0 +1,12 @@
+import { Source, pipe, toPromise, take } from 'wonka';
+import { PromisifiedSource } from '../types';
+
+export function withPromise<T = any>(source$: Source<T>): PromisifiedSource<T> {
+  (source$ as PromisifiedSource<T>).toPromise = () =>
+    pipe(
+      source$,
+      take(1),
+      toPromise
+    );
+  return source$ as PromisifiedSource<T>;
```

Comment:
This cast shouldn't be necessary and it may be better to remove the any fallback